File: sdk/nodejs/src/resources/ipam.js
"use strict";

const { BaseResource } = require("./base");

class IpamResource extends BaseResource {
  async listIps(serviceId) {
    return this._http.get("/api/service/{id}/htcipam/ips");
  }

  async listSubnets(serviceId) {
    return this._http.get("/api/service/{id}/htcipam/subnets");
  }

  async listRdns(serviceId) {
    return this._http.get("/api/service/{id}/htcipam/rdns");
  }

  async setRdns(serviceId, data) {
    return this._http.post("/api/service/{id}/htcipam/rdns", data);
  }

}

module.exports = { IpamResource };

File: sdk/java/src/main/java/vn/hitechcloud/sdk/resources/Ipam.java
package vn.hitechcloud.sdk.resources;

import java.util.Map;
import vn.hitechcloud.sdk.HttpClient;

public class Ipam extends BaseResource {

    public Ipam(HttpClient http) {
        super(http);
    }

    /** Get IPs */
    public Map<String, Object> ips(int id) throws Exception {
        return http.get(String.format("/api/service/%id/htcipam/ips", id));
    }

    /** Get subnets */
    public Map<String, Object> subnets(int id) throws Exception {
        return http.get(String.format("/api/service/%id/htcipam/subnets", id));
    }

    /** Get rDNS */
    public Map<String, Object> rdns(int id) throws Exception {
        return http.get(String.format("/api/service/%id/htcipam/rdns", id));
    }

    /** Set rDNS */
    public Map<String, Object> setRdns(int id, Map<String, Object> data) throws Exception {
        return http.post(String.format("/api/service/%id/htcipam/rdns", id), data);
    }

}
